tool/check-code.sh

```diff
@@ -42,9 +42,14 @@ flutter --version
 if [[ $REFRESH ]]; then
   echo "=> Refreshing code excerpts..."
   (
+    commitHash=$(git reflog -n1 --format='%h')
+    git add .
+    git commit -m "Temporarily commit code excerpt changes"
     set -x
     tool/refresh-code-excerpts.sh --keep-dart-tool
+    git reset --soft $commitHash
   ) || (
+    git --no-pager diff
```

Comment:
Won't this only show the diff if there is an error? I believe if all excerpts were successfully refreshed, it won't result an error, so the diff won't be shown.

Maybe this could be moved directly below the `git reset`? It's probably fine if it always runs.

Reply:
> Won't this only show the diff if there is an error? I believe if all excerpts were successfully refreshed, it won't result an error, so the diff won't be shown.

IIRC, from actual tests, refreshes will throw exceptions if there are codes been refreshed and return `0` if no codes have been refreshed.

If "successfully refreshed" means there are some changes during the refresh and the process ends properly, that is a misunderstanding.

Reply:
I tried it out and the diff doesn't seem to run when excerpts were refreshed. I guess that's because calling refresh script isn't the last executed command in the subshell.

Reply:
I took another look of your suggestion, if all excerpts were refreshed, why the diff is required? The diff flow is only meant to help users to identify not refreshed files.